### PR TITLE
Fix last error with Rails5

### PR DESCRIPTION
**Pull request by n-rodriguez**
This patch fixes the last error with Rails 5.

It generates the right SQL query :

`SELECT "posts".* FROM "posts" INNER JOIN "taggings" ON "posts"."id" = "taggings"."taggable_id" WHERE "taggings"."polytag_id" = 4 AND "taggings"."polytag_type" = 'SpecialTag' AND "taggings"."taggable_type" IN ('Post', 'SpecialPost')`

instead of :

`SELECT "posts".* FROM "posts" INNER JOIN "taggings" ON "posts"."id" = "taggings"."taggable_id" WHERE "taggings"."polytag_id" = 2 AND "taggings"."polytag_type" = 'SpecialTag' AND "taggings"."taggable_type" = 'Post'`

I hope it won't break anything in Rails.

### CODE REVIEW

File: lib/store_base_sti_class_for_5_0.rb [hunk 1]
```diff
@@ -346,6 +346,24 @@ def build_through_record(record)
     end
 
     module Reflection
+      class PolymorphicReflection < ThroughReflection
```
Comment (januszm) [inline lib/store_base_sti_class_for_5_0.rb]: In Rails `5.0.1` this class inherits from `AbstractReflection` , but we're simply **reopening** the class so there is no need to include full definition with inheritance.

Comment (januszm) [inline lib/store_base_sti_class_for_5_0.rb]: @n-rodriguez tests pass with the change above, please amend your commit with:

```
class PolymorphicReflection < AbstractReflection
``` 
Looks good on 5.0.0.1 as well as 5.0.1

Comment (januszm) [reply lib/store_base_sti_class_for_5_0.rb]: @n-rodriguez please disregard outdated comments and simply remove the  ``` < ThroughReflection```
tested.

Comment (n-rodriguez) [reply lib/store_base_sti_class_for_5_0.rb]: Hi @januszm! Thanks for your review. Actually the class inherits from `AbstractReflection` in the [master branch](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/reflection.rb#L978) (which I think will be in Rails 5.1) and from `ThroughReflection` in the [5.0-stable branch](https://github.com/rails/rails/blob/5-0-stable/activerecord/lib/active_record/reflection.rb#L996) but I'll do as you said ;)

**Comment by n-rodriguez** [comment]
Thanks!

**Pull request merged**